packages/react-devtools-shared/src/devtools/views/Components/EditableValue.js

```diff
@@ -67,71 +87,59 @@ export default function EditableValue({
 
       const {key} = event;
 
-      if (key === 'Enter') {
-        if (dataType === 'number') {
-          const parsedValue = parseFloat(editableValue);
-          if (!Number.isNaN(parsedValue)) {
-            overrideValueFn(path, parsedValue);
-          }
-        } else {
-          overrideValueFn(path, editableValue);
+      if (key === 'Enter' && isValid) {
+        const parsedEditableValue = JSON.parse(sanitizeForParse(editableValue));
+
+        if (value !== parsedEditableValue) {
+          overrideValueFn(path, parsedEditableValue);
         }
 
         // Don't reset the pending change flag here.
         // The inspected fiber won't be updated until after the next "inspectElement" message.
         // We'll reset that flag during a subsequent render.
       } else if (key === 'Escape') {
-        setEditableValue(value);
+        setEditableValue(stringifiedValue);
         setHasPendingChanges(false);
+        setIsValid(true);
       }
     },
-    [editableValue, dataType, overrideValueFn, path, value],
+    [editableValue, isValid, dataType, overrideValueFn, path, value],
   );
 
-  // Render different input types based on the dataType
-  let type = 'text';
-  if (dataType === 'boolean') {
-    type = 'checkbox';
-  } else if (dataType === 'number') {
```

Comment:
We don't need this anymore. Now we have the ability to change the value from a number to an array for example.

Reply:
Is it expected that once a prop becomes a boolean, you can't change it back?
![aaaaaKapture 2019-09-09 at 16 42 55](https://user-images.githubusercontent.com/29597/64573687-17e86b00-d321-11e9-8282-04eb46f98f02.gif)

Reply:
I thought it would be cool to keep this as it is. 
Users can still override it via Add new prop -> type `name` and set the value to `"hello world"` for example.
Do you think it would be better if we just displayed `true` / `false` instead of the checkbox? 
That way we can change it more easily.

Reply:
> Users can still override it via Add new prop -> type name and set the value to "hello world" for example.

That seems pretty non-obvious though. (I don't think it would occur to people to do this).

> Do you think it would be better if we just displayed true / false instead of the checkbox?

Yeah 😄 I think we should do this.

Reply:
I'll just go ahead and make this change.